src/main/java/cam72cam/immersiverailroading/tile/TileRailBase.java

```diff
@@ -664,60 +685,65 @@ public void update() {
 		EntityMoveableRollingStock stock;
 		IFluidHandler stock_fluid;
 		IItemHandler stock_items;
+		
+		boolean canTransfer = false;
+		if (this.loaderMode == LoaderMode.DEFAULT_ON && !this.isPowered) canTransfer = true;
+		else if (this.loaderMode == LoaderMode.DEFAULT_OFF && this.isPowered) canTransfer = true;
+		else if (this.loaderMode == LoaderMode.ALWAYS_ON) canTransfer = true;
 
 		try {
 			switch (this.augment) {
 			case ITEM_LOADER:
 				stock = this.getStockNearBy(item_cap);
-				if (stock == null) {
-					break;
-				}
-				stock_items = stock.getCapability(item_cap, null);
-				for (IItemHandler neighbor : getCapsNearby(item_cap)) {
-					transferAllItems(neighbor, stock_items, 1);
+				if (stock == null) break;
```

Comment:
Please leave this as a multiline if.  It is easier to read and fits with the style of the rest of the IR code